crates/ide_assists/src/handlers/convert_tuple_struct_to_named_struct.rs

```diff
@@ -0,0 +1,345 @@
+use hir::{Adt, ModuleDef};
+use ide_db::defs::Definition;
+use syntax::{
+    ast::{self, AstNode, GenericParamsOwner, VisibilityOwner},
+    match_ast,
+};
+
+use crate::{assist_context::AssistBuilder, AssistContext, AssistId, AssistKind, Assists};
+
+// Assist: convert_tuple_struct_to_named_struct
+//
+// Converts tuple struct to struct with named fields.
+//
+// ```
+// struct Inner;
+// struct A$0(Inner);
+// ```
+// ->
+// ```
+// struct Inner;
+// struct A { field1: Inner }
```

Comment:
does this method also work with tuples with more than one item? if so it might be good to include a larger example?

Reply:
yes it supports more than one item, i added a larger example :+1: also spotted a bug while adding a larger example :smile: